### PR TITLE
fix(server): filter warning messages from sessions

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
This fixes the backend filtering for sessions

### CODE REVIEW

**Review by leafty — CHANGES_REQUESTED** [review]
I think I would rather have the following logic for filtering the status field:

```ts
   const cleanedServerEntries = Object.entries(servers).map(([key, session]) => {
    const { details, message, readyNumContainers, state, totalNumContainers } =
      session.status;
    const cleanedStatus = {
      details: details ?? [],
      ...(message ? { message } : {}),
      readyNumContainers: readyNumContainers ?? -1,
      state: state ?? { pod_name: "" },
      totalNumContainers: totalNumContainers ?? -1,
    };
    return [key, { status: cleanedStatus }] as const;
  });
  const cleanedServers = cleanedServerEntries.reduce(
    (obj, [key, value]) => ({ ...obj, [key]: value }),
    {} as Record<string, Session>
  );
```

File: server/src/websocket/handlers/sessions.ts [hunk 1]
```diff
@@ -62,10 +73,30 @@ function heartbeatRequestSessionStatus(
     .then((response) => {
       const statusFetched = response as unknown as SessionsResult;
       const servers = statusFetched?.servers ?? {};
+
+      // Keep only relevant status information, without warning messages
       const cleanStatus: Record<string, Session> = {};
-      // only keep status information
       Object.keys(servers).map((key) => {
-        cleanStatus[key] = { status: servers[key].status };
+        try {
```
[the remark this answers is not in the document]
Comment (lorenzo-cavazzi) [reply server/src/websocket/handlers/sessions.ts:80]: `Object.keys(servers[key].status)`, but it really shouldn't... It was just for safety cause an error here would be bad

File: server/src/websocket/handlers/sessions.ts [hunk 2]
```diff
@@ -62,10 +73,30 @@ function heartbeatRequestSessionStatus(
...
+          const initialStatus: Session["status"] = {
+            details: [],
+            readyNumContainers: -1,
+            state: {
+              pod_name: "",
+            },
+            totalNumContainers: -1,
+          };
+          const filedsToKeep = Object.keys(servers[key].status).filter(
```
[the remark this answers is not in the document]
Comment (lorenzo-cavazzi) [reply server/src/websocket/handlers/sessions.ts]: Fixed in d3c202d212decdd280930ab447aaad43cd2c963e to adjust also line 92 using the variable

**Comment by lorenzo-cavazzi** [comment]
> I think I would rather have the following logic for filtering the status field:
> 
> ```ts
>    const cleanedServerEntries = Object.entries(servers).map(([key, session]) => {
>     const { details, message, readyNumContainers, state, totalNumContainers } =
>       session.status;
>     const cleanedStatus = {
>       details: details ?? [],
>       ...(message ? { message } : {}),
>       readyNumContainers: readyNumContainers ?? -1,
>       state: state ?? { pod_name: "" },
>       totalNumContainers: totalNumContainers ?? -1,
>     };
>     return [key, { status: cleanedStatus }] as const;
>   });
>   const cleanedServers = cleanedServerEntries.reduce(
>     (obj, [key, value]) => ({ ...obj, [key]: value }),
>     {} as Record<string, Session>
>   );
> ```



> 

It looks quite similar to me.
Feel free to push that change if you think it's easier to read :shrug:

**Comment by lorenzo-cavazzi** [comment]
Also, feel free to merge the fix as soon as you are happy with the code

**Pull request merged**